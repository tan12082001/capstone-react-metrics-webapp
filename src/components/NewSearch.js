import { useState } from 'react';
import { useDispatch } from 'react-redux';
import { GetIcon } from './Icons';
import { fetchEarthquakesData, setMinMagnitude } from '../redux/items/itemsSlice';
import '../styles/Search.css';

const NewSearch = () => {
  const [minMag, setMinMag] = useState(8);
  const dispatch = useDispatch();

  const handleSearch = () => {
    dispatch(setMinMagnitude(minMag));
    dispatch(fetchEarthquakesData());
  };
  return (
    <div className="search-outer">
      <div className="search-info">
        <h3>
          Get
          <i> Major </i>
          {' '}
          and
          <i> Strong </i>
          {' '}
          Earthquakes happened between
          <i> 2001-01-01 </i>
          {' '}
          to
          <i> 2023-01-01 </i>
        </h3>
        <span>Major EarthQuakes: 7.0 - 7.9</span>
        <span>Great EarthQuakes: 8.0 - higher</span>
      </div>
      <div className="search-drop">
        <select className="mag-select" value={minMag} onChange={(e) => setMinMag(e.target.value)}>
          <option value={7.0}>7.0 (5 points)</option>
          <option value={7.1}>7.1 (6 points)</option>
          <option value={7.2}>7.2 (7 points)</option>
          <option value={7.3}>7.3 (8 points)</option>
          <option value={7.4}>7.4 (9 points)</option>
          <option value={7.5}>7.5 (10 points)</option>
          <option value={7.6}>7.6 (11 points)</option>
          <option value={7.7}>7.7 (12 points)</option>
          <option value={7.8}>7.8 (13 points)</option>
          <option value={7.9}>7.9 (14 points)</option>
          <option value={8.0}>8.0 or higher (15 points)</option>
          <option value={9.0}>9.0 (higher)</option>
        </select>
        <button type="button" className="search-button" onClick={handleSearch}>
          <GetIcon />
        </button>
      </div>
    </div>
  );
};

export default NewSearch;
