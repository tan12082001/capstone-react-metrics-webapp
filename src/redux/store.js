import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import itemsReducer from './items/itemsSlice';

const store = configureStore({
  reducer: {
    items: itemsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
