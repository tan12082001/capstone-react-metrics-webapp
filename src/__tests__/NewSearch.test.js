import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import NewSearch from '../components/NewSearch';

jest.mock('../components/Icons', () => ({
  GetIcon: () => <button type="button">Mocked GetIcon</button>,
}));

test('NewSearch component render', () => {
  const tree = render(<Provider store={store}><NewSearch /></Provider>);
  expect(tree).toMatchSnapshot();
  expect(tree.getByText('Major EarthQuakes: 7.0 - 7.9')).toBeVisible();
  expect(tree.getByText('7.0 (5 points)')).toBeVisible();
});
