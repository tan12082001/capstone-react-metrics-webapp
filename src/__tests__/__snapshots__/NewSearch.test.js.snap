// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewSearch component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="search-outer"
      >
        <div
          class="search-info"
        >
          <h3>
            Get
            <i>
               Major 
            </i>
             
            and
            <i>
               Strong 
            </i>
             
            Earthquakes happened between
            <i>
               2001-01-01 
            </i>
             
            to
            <i>
               2023-01-01 
            </i>
          </h3>
          <span>
            Major EarthQuakes: 7.0 - 7.9
          </span>
          <span>
            Great EarthQuakes: 8.0 - higher
          </span>
        </div>
        <div
          class="search-drop"
        >
          <select
            class="mag-select"
          >
            <option
              value="7"
            >
              7.0 (5 points)
            </option>
            <option
              value="7.1"
            >
              7.1 (6 points)
            </option>
            <option
              value="7.2"
            >
              7.2 (7 points)
            </option>
            <option
              value="7.3"
            >
              7.3 (8 points)
            </option>
            <option
              value="7.4"
            >
              7.4 (9 points)
            </option>
            <option
              value="7.5"
            >
              7.5 (10 points)
            </option>
            <option
              value="7.6"
            >
              7.6 (11 points)
            </option>
            <option
              value="7.7"
            >
              7.7 (12 points)
            </option>
            <option
              value="7.8"
            >
              7.8 (13 points)
            </option>
            <option
              value="7.9"
            >
              7.9 (14 points)
            </option>
            <option
              value="8"
            >
              8.0 or higher (15 points)
            </option>
            <option
              value="9"
            >
              9.0 (higher)
            </option>
          </select>
          <button
            class="search-button"
            type="button"
          >
            <button
              type="button"
            >
              Mocked GetIcon
            </button>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="search-outer"
    >
      <div
        class="search-info"
      >
        <h3>
          Get
          <i>
             Major 
          </i>
           
          and
          <i>
             Strong 
          </i>
           
          Earthquakes happened between
          <i>
             2001-01-01 
          </i>
           
          to
          <i>
             2023-01-01 
          </i>
        </h3>
        <span>
          Major EarthQuakes: 7.0 - 7.9
        </span>
        <span>
          Great EarthQuakes: 8.0 - higher
        </span>
      </div>
      <div
        class="search-drop"
      >
        <select
          class="mag-select"
        >
          <option
            value="7"
          >
            7.0 (5 points)
          </option>
          <option
            value="7.1"
          >
            7.1 (6 points)
          </option>
          <option
            value="7.2"
          >
            7.2 (7 points)
          </option>
          <option
            value="7.3"
          >
            7.3 (8 points)
          </option>
          <option
            value="7.4"
          >
            7.4 (9 points)
          </option>
          <option
            value="7.5"
          >
            7.5 (10 points)
          </option>
          <option
            value="7.6"
          >
            7.6 (11 points)
          </option>
          <option
            value="7.7"
          >
            7.7 (12 points)
          </option>
          <option
            value="7.8"
          >
            7.8 (13 points)
          </option>
          <option
            value="7.9"
          >
            7.9 (14 points)
          </option>
          <option
            value="8"
          >
            8.0 or higher (15 points)
          </option>
          <option
            value="9"
          >
            9.0 (higher)
          </option>
        </select>
        <button
          class="search-button"
          type="button"
        >
          <button
            type="button"
          >
            Mocked GetIcon
          </button>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
